pandas/core/ops.py

```diff
@@ -658,6 +649,10 @@ def wrapper(left, right, name=name, na_op=na_op):
                                     index=left.index, name=res_name,
                                     dtype=result.dtype)
 
+        elif is_categorical_dtype(left):
```

Comment:
tests that hit this

Reply:
Is the request for comments in the code for what tests hit this path?  Or confirmation here that such tests exist?

Reply:
tests.categorical.test_operators.TestCategoricalOps.test_numeric_like_ops hits this path.  The change here is catching this case early (and explicitly instead of in the `else:` clause within the na_op above (see point where this PR changes `elif isinstance(x, np.ndarray)` to `else: assert isinstance(x, np.ndarray)`

Reply:
my requests are about things that are being added. e.g. if its just refactoing then existing tests are ok, but some things look like they are catching additional cases, so should have tests for these. a coverage analysis can test you (compare before and after)

Reply:
> if its just refactoing then existing tests are ok

This is very nearly pure refactoring to catch things earlier, more explicitly, and with fewer levels of indentation.  I'll go through the diff and annotate any exceptions to this rule.  The `is_categorical_dtype` check on 652 replaces the `else` previously on 621 (i.e. not an exception to the rule).